packages/cli/src/commands/apps/open.ts

```diff
@@ -0,0 +1,32 @@
+import {Args} from '@oclif/core'
+import {Command, flags} from '@heroku-cli/command'
+import * as Heroku from '@heroku-cli/schema'
+import * as open from 'open'
+
+export default class AppsOpen extends Command {
+  static description = 'open the app in a web browser'
+  static topic = 'apps'
+  static aliases = ['apps:open', 'open']
+
+  static examples = [
+    '$ heroku open -a myapp',
+    '$ heroku open -a myapp /foo',
+  ]
+
+  static flags = {
+    app: flags.app({required: true}),
+  }
+
+  static args = {
+    path: Args.string({required: false}),
+  }
+
+  async run() {
+    const {flags, args} = await this.parse(AppsOpen)
+
+    const appResponse = await this.heroku.get<Heroku.App>(`/apps/${flags.app}`)
+    const app = appResponse.body
+    const url = args.path ? `${app.web_url}${args.path}` : `${app.web_url}`
```

Comment:
`url.resolve()` has been deprecated so I updated the logic for that string building here.

Reply:
Is there a reason you did not use the [recommended WHATWG URL API alternative](https://nodejs.org/api/url.html#urlresolvefrom-to)? While the string concatenation you're doing here should be fine in this context as these are both values we're getting from the API, it might be good to have a shared `resolveURL` function that we could use consistently for any URL concatenation operations we need.

Also, if there is the possibility that `args.path` will be falsy and we could open the browser with only the `web_url`, could you please add a test for this behavior?

Reply:
Yeah, initially there looked to be some testing issues using this method, but I was able to resolve them on subsequent commits. Added the additional test.